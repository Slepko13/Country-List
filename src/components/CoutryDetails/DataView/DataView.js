import React from 'react';
import './DataView.scss';
import PropTypes from "prop-types";

const DataView = (props) => {
    let {title, data, position, addictions, maxWidth} = props;
    data = data === "" ? "n/a" : data === "Kiev" ? "Kyiv" : data;
    return (
        <div className={position ?
            `DataView ${position}` :
            "DataView"}
             data-testid="position"
        >
            <div className="info__title">{title}</div>
            <div className="info__content" style={{maxWidth}}>{data}<span style={{marginLeft: "10px"}}>{addictions}</span></div>
        </div>
    )
}
export default DataView;

DataView.propTypes = {
    title: PropTypes.string,
    data: PropTypes.string,
    position: PropTypes.string,
    addictions: PropTypes.string,
    maxWidth: PropTypes.string
}