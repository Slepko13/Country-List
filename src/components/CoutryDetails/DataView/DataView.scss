.DataView {
  margin-left: 20px;
  z-index: 5;
  @media screen and (max-width: 991px) {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin-bottom: 20px;

  }
    .info__title{
      font-family: "Roboto";
      font-style: normal;
      font-weight: 300;
      font-size: 14px;
      line-height: 16px;
      text-align: center;

    }
    .info__content {
      font-family: "Roboto";
      font-style: normal;
      font-weight: bold;
      text-align: center;
      color: #313131;
      line-height: 30px;
      font-size: 26px;
      max-height: 30px;
      text-overflow:ellipsis;
      overflow: hidden;
      max-width: 200px;

      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }
    }
  }

