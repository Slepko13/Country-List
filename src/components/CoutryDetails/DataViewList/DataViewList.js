import React from 'react';
import './DataViewList.scss';

import SeeMore from "../SeeMore/SeeMore";


const DataViewList = (props) =>{

    let{title, data, position, addictions, direction} = props;
    let myData, restData;

    if(!data)    return  <div className={position ? `DataViewList + ${position} `: "DataViewList"}>
        <div className="absent">Data is absent today</div>
    </div>;

        if(data.length <= 3 && data.length > 0 ) {
            myData = [...data];
        }else if(data.length > 3) {
            myData = data.slice(0,3);
            restData = data.slice(3);
        }

    return (
        <div className={position ? `DataViewList + ${position} `: "DataViewList"}>
            <div className="info__title">{title}</div>
            <div className="info__content">{
                data.length ?
                    myData.filter(item=>item.name!=="null")
                        .map(item=>
                        <div
                            className="item"
                            key={item.name}
                        >{addictions}
                                <span className="item__symbol">{
                            !item.symbol ? null : item.symbol ==="null" ? "n/a": `"${item.symbol}"`
                                }</span>
                            {item.name.replace("UTC","GMT")}

                        </div>)
                :
                    <div className="item">n/a</div>
                    }
                {restData ?
                    <SeeMore
                        data={restData}
                        bottom=""
                        top="100%"
                        left=""
                        right=""
                    />: null}

            </div>
            {data.length > 3 ?<div style={{height:"20px"}}></div>:null}

        </div>
    )
}

export default DataViewList;