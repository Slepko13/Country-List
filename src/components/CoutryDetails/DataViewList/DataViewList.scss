.DataViewList{
  margin-right: 10px;
  position: relative;
  border : 2px solid red;

  .info__title{
    font-family: "Roboto";
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
    margin-bottom: 20px;

  }
  .info__content {
    font-family: "Roboto";
    font-style: normal;
    font-weight: bold;
    text-align: center;
    color: #313131;
    display: flex;
    flex-direction: column;


    .item {
      text-align: left;
      line-height: 30px;
      font-size: 26px;
      max-height: 28px;
      text-overflow:ellipsis;
      overflow: hidden;
      max-width: 300px;
      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }


      .item__symbol {
        margin-right: 10px;
      }
    }
  }
}