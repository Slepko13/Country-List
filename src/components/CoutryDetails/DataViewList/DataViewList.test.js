import React from "react";
import DataViewList from "./DataViewList";
import DataView from "../DataView/DataView";


describe("DataViewList component", () => {
    describe("data is null", ()=>{
        const props = {
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewList {...props}/>);
        it("should renders 2 items ", () => {
            expect(component.find('div.item')).toHaveLength(0);
        });

    });
    describe("data length <= 3", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewList {...props}/>);
        it("should renders 2 items ", () => {
            expect(component.find('div.item')).toHaveLength(2);
        });
        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });
    });
    describe(" data length > 3",() => {
        const props = {
            data : [
                {name :"english" , symnol: "e"},
                {name :"french" , symnol: "f"},
                {name :"german", symnol: "g" },
                {name :"polish", symnol: "p" }
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewList {...props}/>);

        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });

        it("should renders 3 items and SeeMore ", () => {
            expect(component.find('div.item')).toHaveLength(3);
            expect(component.find('SeeMore')).toHaveLength(1);
        });

        it("should renders symbol ", () => {
            expect(component.find('span.item__symbol')).toHaveLength(3);
        });

    });
});

