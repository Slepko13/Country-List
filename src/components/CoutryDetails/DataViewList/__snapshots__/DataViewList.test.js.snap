// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataViewListMobile component  data length > 3 should renders properly  1`] = `
<div
  className="DataViewList + left "
>
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
    >
      <span
        className="item__symbol"
      />
      english
    </div>
    <div
      className="item"
      key="french"
    >
      <span
        className="item__symbol"
      />
      french
    </div>
    <div
      className="item"
      key="german"
    >
      <span
        className="item__symbol"
      />
      german
    </div>
    <SeeMore
      bottom=""
      data={
        Array [
          Object {
            "name": "polish",
            "symnol": "p",
          },
        ]
      }
      left=""
      right=""
      top="100%"
    />
  </div>
  <div
    style={
      Object {
        "height": "20px",
      }
    }
  />
</div>
`;

exports[`DataViewListMobile component data length <= 3 should renders properly  1`] = `
<div
  className="DataViewList + left "
>
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
    >
      <span
        className="item__symbol"
      />
      english
    </div>
    <div
      className="item"
      key="french"
    >
      <span
        className="item__symbol"
      />
      french
    </div>
  </div>
</div>
`;
